streampark-spark/streampark-spark-client/streampark-spark-client-api/src/main/scala/org/apache/streampark/spark/client/conf/SparkConfiguration.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.streampark.spark.client.conf
+
+object SparkConfiguration {
+  val defaultParameters = Map[String, Any](
+    "spark.driver.cores" -> "1",
```

Comment:
Can these parameters be customized by users?

Reply:
Yes. Users can customize parameters through `DynamicPropeties`. The user-defined parameters will override the default parameters before the job is submitted.

```java
  //streampark-spark/streampark-spark-client/streampark-spark-client-core/src/main/scala/org/apache/streampark/spark/client/impl/YarnApplicationClient.scala#setDynamicProperties
  private def setDynamicProperties(sparkLauncher: SparkLauncher, properties: Map[String, Any]): Unit = {
    logger.info("Spark launcher start configuration.")
    val finalProperties: Map[String, Any] = SparkConfiguration.defaultParameters ++ properties
    for ((k, v) <- finalProperties) {
      if (k.startsWith("spark.")) {
        sparkLauncher.setConf(k, v.toString)
      } else {
        logger.info("\"{}\" doesn't start with \"spark.\". Skip it.", k)
      }
    }
  }
```

Reply:
good job